### PR TITLE
image_optim: add oxipng, fix optional workers

### PR DESCRIPTION
**Pull request by kilianar**
###### Description of changes

Support for [oxipng](https://github.com/shssoichiro/oxipng) was added with [version 0.31.0](https://github.com/toy/image_optim/blob/master/CHANGELOG.markdown#v0310-2021-10-03). So we add `oxipng` as an optional dependency.

If some optional workers are disabled `image_optim` fails with an error message like:
```
Bin resolving errors:
pngout worker: `pngout` not found; please provide proper binary or disable this worker (--no-pngout argument or `:pngout => false` through options)
```
To fix the problem the flag `--no-program` is added while wrapping if the worker `program` is disabled.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by veprbl** [comment]
@GrahamcOfBorg eval

File: pkgs/applications/graphics/image_optim/default.nix [hunk 1]
```diff
@@ -21,12 +22,27 @@ let
...
+    ++ optional withOxipng oxipng
     ++ optional withJhead jhead
     ++ optional withJpegoptim jpegoptim
     ++ optional withJpegrecompress jpeg-archive
     ++ optional withJpegtran libjpeg
     ++ optional withGifsicle gifsicle
     ++ optional withSvgo svgo;
+
+  disabledWorkersFlags = []
+    ++ optional (!withPngcrush) "--no-pngcrush"
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/image_optim/default.nix:34]:
```suggestion
  disabledWorkersFlags = optional (!withPngcrush) "--no-pngcrush"
```

File: pkgs/applications/graphics/image_optim/default.nix [hunk 2]
```diff
@@ -39,13 +55,14 @@ bundlerApp {
...
-      --prefix PATH : ${lib.escapeShellArg (makeBinPath optionalDepsPath)}
+      --prefix PATH : ${lib.escapeShellArg (makeBinPath optionalDepsPath)} \
+      --add-flags "${lib.concatStringsSep " " disabledWorkersFlags}"
   '';
 
   passthru.updateScript = bundlerUpdateScript "image_optim";
 
   meta = with lib; {
-    description = "Command line tool and ruby interface to optimize (lossless compress, optionally lossy) jpeg, png, gif and svg images using external utilities (advpng, gifsicle, jhead, jpeg-recompress, jpegoptim, jpegrescan, jpegtran, optipng, pngcrush, pngout, pngquant, svgo)";
+    description = "Command line tool and ruby interface to optimize (lossless compress, optionally lossy) jpeg, png, gif and svg images using external utilities (advpng, gifsicle, jhead, jpeg-recompress, jpegoptim, jpegrescan, jpegtran, optipng, oxipng, pngcrush, pngout, pngquant, svgo)";
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/image_optim/default.nix:65]: Please split this into longDescription

**Review by nicknovitski — APPROVED** [review]

File: pkgs/applications/graphics/image_optim/default.nix [hunk 3]
```diff
@@ -15,18 +16,31 @@
 with lib;
 
 let
-  optionalDepsPath = []
-    ++ optional withPngcrush pngcrush
+  optionalDepsPath = optional withPngcrush pngcrush
     ++ optional withPngout pngout
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/image_optim/default.nix:20]: Maybe we should set this off by default to beable to build the package on ofborg.

**Comment by kilianar** [comment]
@SuperSandro2000 I have attended to all the feedback you kindly offered. I believe this pull request is now ready for merging. Could you take a second look at it, please?

**Comment by kilianar** [comment]
Bumped to version [0.31.3](https://github.com/toy/image_optim/releases/tag/v0.31.3)

**Review by nicknovitski — APPROVED** [review]

**Pull request merged**